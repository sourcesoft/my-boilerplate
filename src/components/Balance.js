import React, { Component } from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  padding: 20px 30px;
  width: 100%;
`;
const Key = styled.span`
  font-size: 1.7rem;
`;
const Value = styled.span`
  font-size: 1.5rem;
  margin-left: 35px;
`;

export class Balance extends Component {
  render() {
    return (
      <Wrapper>
        <div>
          <Key>USD</Key>
          <Value>{this.props.usd.toFixed(2)}</Value>
        </div>
        <div>
          <Key>BTC</Key>
          <Value>{this.props.btc.toFixed(8)}</Value>
        </div>
      </Wrapper>
    );
  }
}

export default Balance;
