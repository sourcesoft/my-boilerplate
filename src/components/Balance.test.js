import React from 'react';
import Balance from './Balance.js';
import { shallow } from 'enzyme';

test('should render Balance component with 23 usd and 40 btc', () => {
  const comp = shallow(<Balance usd={23} btc={40} />);
  expect(comp).toMatchSnapshot();
});
