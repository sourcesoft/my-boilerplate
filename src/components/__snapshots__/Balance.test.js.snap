// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Balance component with 23 usd and 40 btc 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Balance
    btc={40}
    usd={23}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div>
          <styled.span>
            USD
          </styled.span>
          <styled.span>
            23.00
          </styled.span>
        </div>,
        <div>
          <styled.span>
            BTC
          </styled.span>
          <styled.span>
            40.00000000
          </styled.span>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <styled.span>
              USD
            </styled.span>,
            <styled.span>
              23.00
            </styled.span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "USD",
            },
            "ref": null,
            "rendered": "USD",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "23.00",
            },
            "ref": null,
            "rendered": "23.00",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <styled.span>
              BTC
            </styled.span>,
            <styled.span>
              40.00000000
            </styled.span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "BTC",
            },
            "ref": null,
            "rendered": "BTC",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "40.00000000",
            },
            "ref": null,
            "rendered": "40.00000000",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div>
            <styled.span>
              USD
            </styled.span>
            <styled.span>
              23.00
            </styled.span>
          </div>,
          <div>
            <styled.span>
              BTC
            </styled.span>
            <styled.span>
              40.00000000
            </styled.span>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <styled.span>
                USD
              </styled.span>,
              <styled.span>
                23.00
              </styled.span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "USD",
              },
              "ref": null,
              "rendered": "USD",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "23.00",
              },
              "ref": null,
              "rendered": "23.00",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <styled.span>
                BTC
              </styled.span>,
              <styled.span>
                40.00000000
              </styled.span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "BTC",
              },
              "ref": null,
              "rendered": "BTC",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "40.00000000",
              },
              "ref": null,
              "rendered": "40.00000000",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
